src/App.tsx

```diff
@@ -1,24 +1,149 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import * as todosService from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+// import { ErrorNotification } from './components/ErrorNotification';
+import { Errors } from './types/Errors';
 
-const USER_ID = 0;
+const USER_ID = 12037;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterValue, setFilterValue] = useState('all');
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.NULL);
+  const [tempTodo, setTempTodo] = useState<Omit<Todo, 'id'> | null>(null);
+
+  useEffect(
+    () => {
+      todosService.getTodos(USER_ID)
+        .then(setTodos)
+        .catch(() => setErrorMessage(Errors.LOAD))
+        .finally(() => setTimeout(() => setErrorMessage(Errors.NULL), 3000));
+    }, [],
+  );
+
+  const todosToRender = useMemo(
+    () => {
+      return todos.filter(todo => {
+        return filterValue === 'all'
+          || (filterValue === 'completed' ? todo.completed : !todo.completed);
+      });
+    },
+    [todos, filterValue],
+  );
+
+  const addTodo = async (inputValue: string) => {
+    const data = {
+      id: 0,
+      title: inputValue,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo(data);
+
+    try {
+      const createdTodo = await todosService.createTodo(data);
+
+      return (
+        setTodos(currentTodos => [...currentTodos, createdTodo]));
+    } catch (error) {
+      setErrorMessage(Errors.ADD);
+      throw error;
+    } finally {
+      setTimeout(() => setErrorMessage(Errors.NULL), 3000);
```

Comment:
In every API call, you set an error and then remove it in 3 seconds. Create a function for it, for example:
```typescript
const handleError = (errorMessage: Errors) => {
  setErrorMessage(errorMessage);
  setTimeout(() => setErrorMessage(Errors.NULL), 3000);
};
```